contracts/asset-proxy/contracts/src/bridges/DexForwarderBridge.sol

```diff
@@ -0,0 +1,210 @@
+import "@0x/contracts-erc20/contracts/src/interfaces/IERC20Token.sol";
+import "@0x/contracts-erc20/contracts/src/LibERC20Token.sol";
+import "@0x/contracts-exchange-libs/contracts/src/IWallet.sol";
+import "@0x/contracts-exchange-libs/contracts/src/LibMath.sol";
+import "@0x/contracts-utils/contracts/src/LibBytes.sol";
+import "@0x/contracts-utils/contracts/src/LibSafeMath.sol";
+import "../interfaces/IERC20Bridge.sol";
+
+
+// solhint-disable space-after-comma, indent
+contract DexForwarderBridge is
+    IERC20Bridge,
+    IWallet
+{
+    using LibSafeMath for uint256;
+
+    /// @dev Data needed to reconstruct a bridge call.
+    struct BridgeCall {
+        address target;
+        uint256 inputTokenAmount;
+        uint256 outputTokenAmount;
+        bytes bridgeData;
+    }
+
+    /// @dev Intermediate state variables used by `bridgeTransferFrom()`, in
+    ///      struct form to get around stack limits.
+    struct TransferFromState {
+        address inputToken;
+        uint256 initialInputTokenBalance;
+        uint256 callInputTokenAmount;
+        uint256 callOutputTokenAmount;
+        uint256 totalInputTokenSold;
+        BridgeCall[] calls;
+    }
+
+    event DexForwarderBridgeCallFailed(
+        address indexed target,
+        address inputToken,
+        address outputToken,
+        uint256 inputTokenAmount,
+        uint256 outputTokenAmount
+    );
+
+    /// @dev Executes a series of calls, forwarding .
+    /// @param outputToken The token being bought.
+    /// @param to The recipient of the bought tokens.
+    /// @param bridgeData The abi-encoeded input token address.
+    /// @return success The magic bytes if successful.
+    function bridgeTransferFrom(
+        address outputToken,
+        address /* from */,
+        address to,
+        uint256 /* amount */,
+        bytes calldata bridgeData
+    )
+        external
+        returns (bytes4 success)
+    {
+        TransferFromState memory state;
+        (
+            state.inputToken,
+            state.calls
+        ) = abi.decode(bridgeData, (address, BridgeCall[]));
+
+        state.initialInputTokenBalance = IERC20Token(state.inputToken).balanceOf(address(this));
+
+        for (uint256 i = 0; i < state.calls.length; ++i) {
+            // Stop if the we've sold all our input tokens.
+            if (state.totalInputTokenSold >= state.initialInputTokenBalance) {
+                break;
+            }
+
+            BridgeCall memory call = state.calls[i];
+            // Compute token amounts.
+            state.callInputTokenAmount = LibSafeMath.min256(
+                call.inputTokenAmount,
+                state.initialInputTokenBalance.safeSub(state.totalInputTokenSold)
+            );
+            state.callOutputTokenAmount = LibMath.getPartialAmountFloor(
+                state.callInputTokenAmount,
+                call.inputTokenAmount,
+                call.outputTokenAmount
+            );
+
+            (bool didSucceed, ) = address(this)
```

Comment:
Maybe we should decode the return data here and check that it equals BRIDGE_SUCCESS?

Reply:
`execteBridgeCall()` doesn't return anything. It just succeeds or reverts.